ci/infra_cccl.sh

```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
+source /opt/devcontainer/bin/update-envvars.sh
```

Comment:
Critical:

I believe this should not be necessary. The goal is to run infra without some env vars, so we should do that in the github workflow where we once build with and once without sccache

Reply:
@miscco I can imagine updating `matrix.yaml` with flag that allows for conditioning these calls to `unset_envvars` on or off in this scrtipt, but I don't see how to avoid having them set here in the first place (and therefore needing to unset them at this point). Can you give me a pointer to whatever mechanism will afford that possibility?

Reply:
But I'd also re-iterate my question above: is the goal to simulate missing `sccache` itself, or to simulate a missing `sccache` *configuration*?

Reply:
@miscco I had hoped to simulate the "missing `sccache` itself" case by using the CMake presets:
```diff
diff --git a/CMakePresets.json b/CMakePresets.json
index 61cb88eca..40b8b1295 100644
--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -321,7 +321,10 @@
       "cacheVariables": {
         "CMAKE_CUDA_COMPILER": "nvcc",
         "CCCL_ENABLE_EXAMPLES": true,
-        "CCCL_ENABLE_TESTING": true
+        "CCCL_ENABLE_TESTING": true,
+        "CMAKE_C_COMPILER_LAUNCHER": " ",
+        "CMAKE_CXX_COMPILER_LAUNCHER": " ",
+        "CMAKE_CUDA_COMPILER_LAUNCHER": " "
       }
     }
   ],
```

Unfortunately, if I rename `sccache` to `sccache.bak` to test this, and do a clean (`rm -rf build/latest/*`) build, then the configure fails because `sccache` is missing, even though you can see the CMake launcher vars are correctly set to the presets values, in the output. 

<details>

```
(cccl) coder ➜ ~/cccl $ rm -rf build/latest/*
(cccl) coder ➜ ~/cccl $ ./ci/infra_cccl.sh 
================== ⚙️ Environment Details ======================
pwd=/home/coder/cccl/ci
BUILD_DIR=/home/coder/cccl/build/cuda12.5-gcc13
CXX_STANDARD=17
CXX=/usr/bin/g++
CUDACXX=/usr/local/cuda/bin/nvcc
CUDAHOSTCXX=/usr/bin/g++
NVCC_VERSION=(undefined)
CMAKE_BUILD_PARALLEL_LEVEL=12
CTEST_PARALLEL_LEVEL=1
CCCL_BUILD_INFIX=cuda12.5-gcc13
GLOBAL_CMAKE_OPTIONS=(undefined)
TBB_ROOT=(undefined)
Current commit is:
commit b036d5d7bd442f6cc7616fe60e18d1f87facea0b (HEAD -> bv/ci-sccache-check, origin/bv/ci-sccache-check)
Author: bryan <bvandeven@nvidia.com>
Date:   Mon Aug 19 15:27:53 2024 -0700

    no un-conditioned calls to sccache
Thu Aug 22 16:27:11 2024       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 555.42.06              Driver Version: 555.42.06      CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Quadro RTX 8000                Off |   00000000:15:00.0 Off |                  Off |
| 33%   38C    P8             31W /  260W |       6MiB /  49152MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
|   1  Quadro RTX 8000                Off |   00000000:2D:00.0  On |                  Off |
| 34%   53C    P8             31W /  260W |    1027MiB /  49152MiB |     17%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
+-----------------------------------------------------------------------------------------+
cmake version 3.30.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
ctest version 3.30.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
================== End ⚙️ Environment Details - Success ==================

================== 🛠️  CMake Configure CCCL Infra ======================
Working directory: /home/coder/cccl
Running command: cmake --preset=cccl-infra --log-level=VERBOSE
Preset CMake variables:

  CCCL_ENABLE_CUB:BOOL="FALSE"
  CCCL_ENABLE_CUDAX:BOOL="FALSE"
  CCCL_ENABLE_EXAMPLES:BOOL="TRUE"
  CCCL_ENABLE_LIBCUDACXX:BOOL="FALSE"
  CCCL_ENABLE_TESTING:BOOL="TRUE"
  CCCL_ENABLE_THRUST:BOOL="FALSE"
  CCCL_ENABLE_UNSTABLE:BOOL="TRUE"
  CMAKE_BUILD_TYPE="Release"
  CMAKE_CUDA_ARCHITECTURES="60;70;80"
  CMAKE_CUDA_COMPILER="nvcc"
  CMAKE_CUDA_COMPILER_LAUNCHER=" "
  CMAKE_CXX_COMPILER_LAUNCHER=" "
  CMAKE_C_COMPILER_LAUNCHER=" "
  CUB_ENABLE_INSTALL_RULES:BOOL="TRUE"
  THRUST_ENABLE_INSTALL_RULES:BOOL="TRUE"
  cudax_ENABLE_INSTALL_RULES:BOOL="TRUE"
  libcudacxx_ENABLE_INSTALL_RULES:BOOL="TRUE"

-- The CXX compiler identification is GNU 13.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ - broken
CMake Error at /usr/share/cmake-3.30/Modules/CMakeTestCXXCompiler.cmake:73 (message):
  The C++ compiler

    "/usr/bin/g++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/home/coder/cccl/build/cuda12.5-gcc13/cccl-infra/CMakeFiles/CMakeScratch/TryCompile-ccEPGR'
    
    Run Build Command(s): /usr/bin/ninja -v cmTC_b2fb9
    [1/2] /usr/bin/sccache /usr/bin/g++    -o CMakeFiles/cmTC_b2fb9.dir/testCXXCompiler.cxx.o -c /home/coder/cccl/build/cuda12.5-gcc13/cccl-infra/CMakeFiles/CMakeScratch/TryCompile-ccEPGR/testCXXCompiler.cxx
    FAILED: CMakeFiles/cmTC_b2fb9.dir/testCXXCompiler.cxx.o 
    /usr/bin/sccache /usr/bin/g++    -o CMakeFiles/cmTC_b2fb9.dir/testCXXCompiler.cxx.o -c /home/coder/cccl/build/cuda12.5-gcc13/cccl-infra/CMakeFiles/CMakeScratch/TryCompile-ccEPGR/testCXXCompiler.cxx
    /bin/sh: 1: /usr/bin/sccache: not found
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:17 (project)


-- Configuring incomplete, errors occurred!
================== End 🛠️  CMake Configure CCCL Infra - Failed - Duration: 0s ==================
```

</details>

I tried setting to empty strings, or spaces, etc. The only way I have been able to get the build not try to use `sccache` is to `unset` the vars in the script as above. Can you provide a concrete suggestion for what you had in mind? I am not sure what else to try here, besides conditioning the `unsets` in the script (or splitting off a separate script for "missing sccache" case).